.github/workflows/generator_generic_slsa3.yml

```diff
@@ -154,9 +163,36 @@ jobs:
           path: "${{ inputs.attestation-name }}"
           sha256: "${{ needs.generator.outputs.attestation-sha256 }}"
 
-      - name: Release
+      - name: Upload provenance
         uses: softprops/action-gh-release@1e07f4398721186383de40550babbdf2b84acfc5 # tag=v0.1.14
         id: release
         with:
           files: |
             ${{ inputs.attestation-name }}
+
+  # upload-url uploads provenance to the release if
+  # `upload-url` is non-empty.
+  upload-url:
+    runs-on: ubuntu-latest
+    needs: [generator]
+    permissions:
+      contents: write
+    if: inputs.upload-url != '' && inputs.upload-assets == true
+    steps:
+      - name: Download the provenance
+        uses: slsa-framework/slsa-github-generator/.github/actions/secure-download-artifact@e178bc7b362c7cb93899a9721da7a0d7d1286869
+        with:
+          name: "${{ inputs.attestation-name }}"
+          path: "${{ inputs.attestation-name }}"
+          sha256: "${{ needs.generator.outputs.attestation-sha256 }}"
+
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # tag=v3.0.2
+      - name: Upload provenance
+        uses: ./.github/actions/upload-assets
```

Comment:
What was the reason we needed to create our own action for this?

Reply:
There is no available and maintained action to do that. It seems to not be a common case. The action that job project uses is https://github.com/actions/upload-release-asset, but it's not maintained. I've just copied their code and updated to latest dependency versions.

Reply:
Since the normal supported wa to upload to releases is to use the `softprops/action-gh-release` and that action has a `tag_name` in order to support this use case, I wonder if we should just accept a `tag-name` input instead of `upload-url` and pass it along to `softprops/action-gh-release` (that we already call)?

Reply:
I fear we are adding a new action that we need to support just because our user jib is using the `upload-url` option on the unsupported action. I'm sure jib could update to use a `tag-name` to specify the release on `workflow_dispatch` events.

Reply:
Great idea. I'll re-work this PR then. Thanks for the suggestion

Reply:
Changes made. PTAL